src/Symfony/Bridge/Doctrine/ManagerRegistry.php

```diff
@@ -24,7 +25,21 @@
  */
 abstract class ManagerRegistry extends AbstractManagerRegistry implements ContainerAwareInterface
 {
-    use ContainerAwareTrait;
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
+    /**
+     * @deprecated since version 3.4, to be removed in 4.0 alongside with the ContainerAwareInterface type.
+     * @final since version 3.4
```

Comment:
can't it just be deprecated?

Reply:
I think so: `@deprecated` doesn't trigger a deprecation when a method is overridden but the parent is not called. `@final` does.